.circleci/config.yml

```diff
@@ -294,244 +267,189 @@ commands:
-          # Example of string before formatting:
-          #   src/test/java/org/pmiops/*********/monitoring/LogsBasedMetricsServiceTest.java src/test/java/org/pmiops/*********/monitoring/MonitoringServiceTest.java
-          # Example of Gradle command after formatting:
-          #   gradle test --tests org.pmiops.*********.monitoring.LogsBasedMetricsServiceTest --tests org.pmiops.*********.monitoring.MonitoringServiceTest
-          #
-          # We exclude the :genomics:test task below because otherwise gradle will attempt to run all the regular API unit tests under the genomics
-          # submodule and will be unable to find them and will barf
-          command: |
-            CLASSNAMES=$(circleci tests glob "src/test/java/**/*Test.java" "src/test/java/**/*Test.kt" \
-              | cut -c 1- \
-              | sed 's@src/test/java/@@' \
-              | sed 's@/@.@g' \
-              | sed 's/\.[^.]*$//' \
-              | circleci tests split --split-by=timings --timings-type=classname --index=$CIRCLE_NODE_INDEX)
-            GRADLE_ARGS=$(echo $CLASSNAMES | awk '{for (i=1; i<=NF; i++) print "--tests", $i }')
-            gradle :test $GRADLE_ARGS
-      - run:
-          # Lint last; it's more important to surface test failures early.
-          name: Java linting
-          working_directory: ~/workbench/api
-          command: |
-            if [ "$CIRCLE_NODE_INDEX" == 0 ]; then
-              ./gradlew spotlessCheck
-            fi
-      - store_test_results:
-           path: ~/workbench/api/build/test-results/test
-      # Save JUnit test results in artifacts because it provides the timing info on tests.
-      - store_artifacts:
-          path: ~/workbench/api/build/test-results/test
-          destination: JunitTestResult
-      - manage_api_cache:
-          save: true
+      - run-api-test:
+          additional_steps:
+            - run:
+                name: Validate Swagger definitions
+                working_directory: ~/workbench/api
+                command: |
+                  if [ "$CIRCLE_NODE_INDEX" == 0 ]; then
+                    ./project.rb validate-swagger --project-prop verboseTestLogging=yes
+                  fi
+            - run:
+                name: Run Java genomics unit tests
+                working_directory: ~/workbench/api
+                command: |
+                  CLASSNAMES=$(circleci tests glob "genomics/src/test/java/**/*Test.java" \
+                    | cut -c 1- \
+                    | sed 's@genomics/src/test/java/@@' \
+                    | sed 's@/@.@g' \
+                    | sed 's/\.[^.]*$//' \
+                    | circleci tests split --split-by=timings --index=$CIRCLE_NODE_INDEX)
+                  GRADLE_ARGS=$(echo $CLASSNAMES | awk '{for (i=1; i<=NF; i++) print "--tests", $i }')
+                  gradle -p genomics test $GRADLE_ARGS
+            - run:
+                name: Run Java api unit tests
+                working_directory: ~/workbench/api
```

Comment:
@calbach At some point soon  I feel like we're going to want to start calling the main backend "api". There are APIs left and right. It's also confusing to see `api` and `genomics` listed as if they're orthogonal concepts, but I don't have much context on that.

Reply:
My opinion on this hasn't changed since it was last discussed: https://github.com/all-of-us/workbench/pull/3658#discussion_r438750326

genomics is a subdir under api - it is split out in CircleCI because the genomics tools gradle project is independent, though that could likely be cleaned up.